Bugzilla/ModPerl/BlockIP.pm

```diff
@@ -40,7 +40,13 @@ sub handler {
     my $r = shift;
     return Apache2::Const::OK if $r->uri =~ $STATIC_URI;
 
-    my $ip = $r->headers_in->{'X-Forwarded-For'} // $r->connection->remote_ip;
+    my $ip = $r->headers_in->{'X-Forwarded-For'};
+    if ($ip) {
+        $ip = (split(/\s*,\s*/ms, $ip))[-1];
```

Comment:
i'm not 100%, but i'm pretty sure you want the second-last ip address here; iirc the last one will always be zeus.

perhaps at this point it's better to use Bugzilla::Util::remote_ip(), which we know does the right thing with respect to x-forwarded-for and inbound proxies (it works backwards through x-forward-for until it finds the first non-proxy address).